### PR TITLE
[Merged by Bors] - feat: Add batch mode for CLI producer

### LINKED ISSUE #915: Update CLI Producer to send records in batches
Since we now have a batch producer client API, we should leverage it in the CLI producer in cases where we are reading directly from files and we have an estimated line count. We should not use it when reading from stdin or piped files, since we do not know when they will end and may not have a chance to flush records before closing.

This adds a new flag to the `fluvio produce` command: `--batch-size=X`, which will allow the user to specify the number of records that should be sent in a single request. The default batch size is 50.

### PR DESCRIPTION
**Pull request by nicholastmosher**
Closes #915 

- When using `fluvio produce topic -f <file>`, the producer will send lines in batches of 50 by defualt
- The `fluvio produce topic --batch-size=X` argument can be used to adjust the size of the batch

### CODE REVIEW

**Comment by nicholastmosher** [comment]
@sehz how does this look?

File: src/cli/src/consumer/produce/mod.rs [hunk 1]
```diff
@@ -56,12 +56,19 @@ impl ProduceOpt {
 
         match &self.file {
             Some(path) => {
-                let mut reader = BufReader::new(File::open(path)?);
-                self.produce_lines(&mut producer, &mut reader).await?;
+                let reader = BufReader::new(File::open(path)?);
+                let lines: Vec<_> = reader.lines().filter_map(|it| it.ok()).collect();
+                let batches: Vec<_> = lines.chunks(50).collect();
```
Comment (sehz) [inline src/cli/src/consumer/produce/mod.rs:61]: I think it is possible to combine `lines`,`batches` into a single iterator

**Review by sehz — CHANGES_REQUESTED** [review]
LGTM.  Couple of nits

File: src/cli/src/consumer/produce/mod.rs (hunk 1 above)
Comment (nicholastmosher) [reply src/cli/src/consumer/produce/mod.rs:61]: I believe that `chunks` is a method on slices, not on iterators. So I needed to collect into a Vec first to make this work

**Review by sehz — CHANGES_REQUESTED** [review]
add to change log

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,15 +1,20 @@
 # Release Notes
 
+## Platform Version 0.8.3 - 2020-05-13
```
Comment (sehz) [inline CHANGELOG.md:3]: Need to rebase to get latest changelog.

Comment (nicholastmosher) [reply CHANGELOG.md]: Ah, got it

**Review by sehz — APPROVED** [review]

**Pull request closed without merging**